index.js

```diff
@@ -1,20 +1,40 @@
-var HalloweenBot = require('./bots/halloween_bot')
+const { GuessBot, startWord, guessWord } = require('./bots/guess_bot.js')
 
-class MyBot extends HalloweenBot {
-  aliveResponse (message) {
-    if (message.text === 'who is alive') {
-      this.send('I am')
+const BANDIT_BOT_ID = '<@U017GNL24JY>'
+
+class MyBot extends GuessBot {
+  constructor () {
+    super()
+    this.banditBotId = BANDIT_BOT_ID
+    this.start()
+  }
+
+  handleExplorationSpace(space, banditBotId) {
+    if (banditBotId) {
+      this.banditBotId = `<@${banditBotId}>`
```

Comment:
Looks like the bot id is declared as a const above -- will this be passed in to this function as well?

Reply:
The reason the bot id has to be a hardcoded const above is that the bandit needs to be mentioned to start a game. However I added this parameter because I think there is some value to having the guess bot know who is signaling them. This allows the guess bot to make decisions in the case where a bot other than the bandit starts the game (althogh in hindsight I didn't go far enough)

To expand a bit more, I think it would be interesting to explore an protocol where the bot running the game (game bot) does _not_ need to be mentioned to start a round, instead relying on a "game word" or words. 

For example, the "game word" for the bandit game could be something like "where is the bandit". Whenever a contestant sends that message, the game bot could start a round for that contestant using the id of the sender. The game bot would then respond and the contestant could also use the id of the sender to mention the game bot in follow up messages.

All that said, I wanted to remain compatible with the bandit in its current form.

Reply:
I added 0361160c3bc76f0f08c3ac22bd0b935f49119396 which passes the bandit bot's id to the reward handler as well. 

In its current form, the "starter kit" implementation will start play a game with anyone who sends a "begin" message to it. I expect people will want to control this a bit more, and with the addition of the id to the reward handler, it is now possible for implementations to decide to whom and how to respond. 

For example, you could implement an allow list of yourself (for testing) and the bandit.